internal/core/processing.go

```diff
@@ -236,12 +243,43 @@ func downloadBookImage(url, dstPath string) error {
 		draw.Draw(bg, bgRect, fg, fgPosition, draw.Over)
 
 		// Save to file
-		err = jpeg.Encode(dstFile, bg, nil)
+		err = jpeg.Encode(tmpFile, bg, nil)
 	}
 
 	if err != nil {
 		return fmt.Errorf("failed to save image %s: %v", url, err)
 	}
 
+	err = MoveToDestination(dstPath, tmpFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func MoveToDestination(dstPath string, tmpFile *os.File) error {
```

Comment:
I would use `MoveFileToDestination` in case we want to have something for folders later.
```suggestion
func MoveFileToDestination(dstPath string, tmpFile *os.File) error {
```

Reply:
Add comment making clear that `dstPath` requires the filename. You can add a third argument for the filename, up to you.